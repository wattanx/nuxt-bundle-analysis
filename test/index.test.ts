import { execSync } from 'child_process';
import fs from 'fs';

describe('ts test', () => {
  it('generated __bundle_analysis.json', async () => {
    expect(
      fs.existsSync('playground/.nuxt/analyze/__bundle_analysis.json')
    ).toBeTruthy();
  });

  it('generated __bundle_analysis_comment.txt', () => {
    expect(
      fs.existsSync('playground/.nuxt/analyze/__bundle_analysis_comment.txt')
    ).toBeTruthy();
  });

  it('comparison results test', () => {
    const contents = fs
      .readFileSync('playground/.nuxt/analyze/__bundle_analysis_comment.txt')
      .toString();

    const splitContents = contents.split('\n');

    // decreased
    expect(splitContents.find((x) => x.includes('app'))).toContain('ðŸŸ¢');

    // increased
    expect(splitContents.find((x) => x.includes('pages/index'))).toContain(
      'ðŸ”´'
    );

    // removed
    expect(splitContents.find((x) => x.includes('pages/test'))).toContain(
      'removed'
    );

    // added
    expect(
      splitContents.find((x) => x.includes('components/nuxt-logo'))
    ).toContain('added');

    expect(contents).toMatchSnapshot();
  });

  it('Snapshot of comparison results', () => {
    const contents = fs
      .readFileSync('playground/.nuxt/analyze/__bundle_analysis_comment.txt')
      .toString();

    expect(contents).toMatchSnapshot();
  });
});
