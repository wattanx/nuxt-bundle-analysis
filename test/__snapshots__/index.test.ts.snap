// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ts test > Snapshot of comparison results 1`] = `
"
<!-- __NUXTJS_BUNDLE -->
# Bundle Size
| Route | Size (gzipped) |
| --- | --- |
| \`app\` | 97.67 kB (ðŸŸ¢ -2.33 kB) |
| \`pages/index\` | removed |
| \`components/tutorial\` | removed |
| \`pages/test\` | removed |"
`;

exports[`ts test > Vite comparison results test 1`] = `
"
<!-- __NUXTJS_BUNDLE -->
# Bundle Size
| Route | Size (gzipped) |
| --- | --- |
| \`app\` | 60.14 kB (ðŸŸ¢ -39.86 kB) |
| \`pages/index\` | removed |
| \`components/tutorial\` | removed |
| \`pages/test\` | removed |"
`;

exports[`ts test > comparison results test 1`] = `
"
<!-- __NUXTJS_BUNDLE -->
# Bundle Size
| Route | Size (gzipped) |
| --- | --- |
| \`app\` | 97.67 kB (ðŸŸ¢ -2.33 kB) |
| \`pages/index\` | removed |
| \`components/tutorial\` | removed |
| \`pages/test\` | removed |"
`;
